include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -376,7 +376,8 @@ namespace internal
                                    : (Eval::dofs_per_cell > Eval::n_q_points ?
                                       Eval::dofs_per_cell : Eval::n_q_points);
     const unsigned int max_stack_size = 100;
-    VectorizedArray<Number>  temp_data[(temp_size > 0 && temp_size < max_stack_size) ? 2*temp_size : 1];
+    AlignedVector<VectorizedArray<Number>>  temp_data;
+    temp_data.resize_fast((temp_size > 0 && temp_size < max_stack_size) ? 2*temp_size : 1);
```

Comment:
Just to clarify: We should simply use the array for temporary data that is provided already, the `scratch_data` input argument.